src/Application/Command/CreateMilestones.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Application\Command;
+
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Laminas\AutomaticReleases\Github\Api\GraphQL\Query\GetGithubMilestone;
+use Laminas\AutomaticReleases\Github\Api\V3\CreateMilestone;
+use Laminas\AutomaticReleases\Github\Api\V3\CreateMilestoneFailed;
+use Laminas\AutomaticReleases\Github\Event\Factory\LoadCurrentGithubEvent;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class CreateMilestones extends Command
+{
+    private LoadCurrentGithubEvent $loadEvent;
+    private GetGithubMilestone $getMilestone;
+    private CreateMilestone $createMilestone;
+    private LoggerInterface $logger;
+
+    public function __construct(
+        LoadCurrentGithubEvent $loadEvent,
+        GetGithubMilestone $getMilestone,
+        CreateMilestone $createMilestone,
+        LoggerInterface $logger
+    ) {
+        parent::__construct('laminas:automatic-releases:create-milestones');
+
+        $this->loadEvent       = $loadEvent;
+        $this->getMilestone    = $getMilestone;
+        $this->createMilestone = $createMilestone;
+        $this->logger          = $logger;
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $milestoneClosedEvent = ($this->loadEvent)();
+        $repositoryName       = $milestoneClosedEvent->repository();
+        $releaseVersion       = $milestoneClosedEvent->version();
+        $milestone            = ($this->getMilestone)($repositoryName, $milestoneClosedEvent->milestoneNumber());
+
+        $milestone->assertAllIssuesAreClosed();
+
+        $this->createMilestoneIfNotExists($repositoryName, $releaseVersion->nextPatch(), $output);
+        $this->createMilestoneIfNotExists($repositoryName, $releaseVersion->nextMinor(), $output);
+        $this->createMilestoneIfNotExists($repositoryName, $releaseVersion->nextMajor(), $output);
+
+        return 0;
+    }
+
+    private function createMilestoneIfNotExists(RepositoryName $repositoryName, SemVerVersion $version, OutputInterface $output): bool
+    {
+        try {
+            ($this->createMilestone)($repositoryName, $version);
+        } catch (CreateMilestoneFailed $e) {
+            $this->logger->info($e->getMessage());
```

Comment:
In PSR-3, an exception should be given as a logging context to the logger: `['exception' => $e]` as added parameter, specifically

Reply:
One other point: in previous usage of the logger, I've been using either the interface name or class name (if not implementing an interface) as a prefix for the message: `[InterfaceOrClassName] `. This makes it far easier to understand **what** is generating the log message.